### ISSUE TITLE
Support $relativeDir as executorMap parameter

### ISSUE THREAD

**Issue by Jiia** [formulahendry/vscode-code-runner#559]
My project is running inside a docker container, and I'd like create a custom execute command to run the file inside the container. However as there is no parameter for relative path I'm unable to build a command that would be able to locate the file inside the container.

The closes I can get is this:
`docker-compose exec -T my-container bash -c 'php -f $fullFileName'`

But it outputs something like

**Solution**
Relative path parameter:

`docker-compose exec -T my-container bash -c 'php -f /container/path/to/project/$relativeDir/$fileName`

**Comment by Jiia** [comment]
Managed to work around it by setting `fileDirectoryAsCwd` to false and using realpath.

```json
"code-runner.executorMap": {
    "php": "docker-compose exec -T --env codeRunnerFile=$(realpath --relative-to=${PWD} $fullFileName) my-container bash -c \"php -d display_errors=1 -d /container/path/toProject/\\${codeRunnerFile}\""
},
"code-runner.fileDirectoryAsCwd": false
```

**Comment by benjaminpinto** [comment]
$relativeDir would be a great feature.

**Comment by palansher** [comment]
yes, I it is reaaly useful to imlement such vars as is in VS CODE:
${relativeFile} - the current opened file relative to workspaceFolder
${relativeFileDirname} - the current opened file's dirname relative to workspaceFolder

Workaround works, thank you!

**Comment by ldeng-ustc** [comment]
It seems that the addition of this variable is still a long way off, so I provide a temporary alternative.
For those who can use shell commands, one way to get the relative path is to use the `realpath` command, for example, `$relativeDir` can be replaced with the following command:

```
`realpath -s --relative-to=$workspaceRoot $dir`
```

For example, I use the following configuration to run cmake to generate executable files under `build` that have the same path structure as the source code:

```json
    "code-runner.executorMap": {
        "cpp": "./build/`realpath -s --relative-to=$workspaceRoot $dir`/$fileNameWithoutExt",
    },
```

Of course, this does not look elegant, but at least it is an optional solution.
